File: app/src/main/java/com/jlbc/bouncingfiguresjava/Figure.java
package com.jlbc.bouncingfiguresjava;

import android.graphics.Canvas;
import android.graphics.RectF;

/*
 *  This interface represents the behaviour of any figure.
 *
 *  METHODS
 *      update
 *      draw
 */
public interface Figure {

    void update(RectF dimensions);
    void draw(Canvas canvas);

}

File: app/src/main/java/com/jlbc/bouncingfigureskotlin/Figure.kt
package com.jlbc.bouncingfigureskotlin

import android.graphics.Canvas
import android.graphics.RectF

/*
 *  This interface represents the behaviour of any figure.
 *
 *  METHODS
 *      update
 *      draw
 */
interface Figure {

    fun update(dimensions: RectF)
    fun draw(canvas: Canvas)

}